arch/Kconfig

```diff
@@ -464,6 +464,9 @@ endmenu
 # Architecture Capabilities
 #
 
+config ARCH_HAS_NO_MULTITHREADING_SUPPORT
```

Comment:
this config sounds and reads wrong. I read it as: `This architecture does not support multi-threading.`

Maybe it should be 'ARCH_HAS_SINGLE_THREAD_SUPPORT' or something along those lines.

Reply:
Ah, that's right, @nashif , it's funny, actually :)
Yeap, your suggestion is fine.

Reply:
Done @nashif @galak